### DISCUSSION TITLE
Is the generateHydrationScript broken in v1.3?

### DISCUSSION THREAD

**Discussion by ghost** [solidjs/solid#786] [General]
This example doesn't work after upgrading to version 1.3: 
https://github.com/brillout/vite-plugin-ssr/tree/master/examples/solid

```
TypeError: Cannot read properties of undefined (reading 'eventNames')
    at Proxy.generateHydrationScript (/share/Downloads/vite-plugin-ssr/examples/solid/node_modules/solid-js/web/dist/server.cjs:576:3)
    at render (/share/Downloads/vite-plugin-ssr/examples/solid/renderer/_default.page.server.tsx:35:55)
    at executeRenderHook (/share/Downloads/vite-plugin-ssr/examples/solid/node_modules/vite-plugin-ssr/node/renderPage.ts:780:20)
    at renderPage (/share/Downloads/vite-plugin-ssr/examples/solid/node_modules/vite-plugin-ssr/node/renderPage.ts:156:34)
    at renderPageWithoutThrowing (/share/Downloads/vite-plugin-ssr/examples/solid/node_modules/vite-plugin-ssr/node/renderPage.ts:211:12)
    at /share/Downloads/vite-plugin-ssr/examples/solid/server/index.ts:30:25
```

It seems the problem is on this line in function generateHydrationScript():
https://github.com/brillout/vite-plugin-ssr/blob/master/examples/solid/renderer/_default.page.server.tsx#L35

**Comment by ryansolid** [comment]
Yeah my bad I see the bug. For now just pass an empty object to `generateHydrationScript` like `generateHydrationScript({})`

I missed defaulting it. All my tests were testing specific options and not the absence of any.

**Comment by ghost** [comment]
@ryansolid sorry, but it's still a mistake if I add {}. Function `dangerouslySkipEscape` does not allow the script generated by `generateHydrationScript({})`.

```
Error: Unable to parse HTML; Incorrectly opened comment.
 at {"file":"/","line":9,"column":543}
7  |          <meta name="description" content="App using Vite + vite-plugin-ssr" />
8  |          <title>Vite SSR app</title>
9  |          <script>var e,t;e=window._$HY||(_$HY={events:[],completed:new WeakSet,r:{}}),t=e=>e&&e.hasAttribute&&(e.hasAttribute("data-hk")?e:t(e.host&&e.host instanceof Node?e.host:e.parentNode)),["click","input"].forEach((o=>document.addEventListener(o,(o=>{let s=o.composedPath&&o.composedPath()[0]||o.target,a=t(s);a&&!e.completed.has(a)&&e.events.push([a,o])})))),e.init=(t,o)=>{e.r[t]=[new Promise(((e,t)=>o=e)),o]},e.set=(t,o,s)=>{(s=e.r[t])&&s[1](o),e.r[t]=[o]},e.unset=t=>{delete e.r[t]},e.load=(t,o)=>{if(o=e.r[t])return o[0]};</script><!xs>
   |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                ^
```
(arrow points to <!xs> at the end of the last line )
If i remove the dangerouslySkipEscape function, i get a broken layout in the browser

**Reply by ryansolid** [reply]
Interesting. That is a shorthand for HTML comments. I wonder if they are running it through an HTML validator or something since browsers know how to handle this and will automatically convert it to `<!--xs-->`. I suppose I could change it to the fully formed comment here.

**Reply by ghost** [reply]
without a `dangerouslySkipEscape` it looks like this in the browser (chrome and FF)
![Screenshot from 2022-01-08 11-22-03](https://user-images.githubusercontent.com/3522914/148628086-8f8f7714-a28d-41fb-8180-7f1f58c5ee32.png)

**Reply by ryansolid** [reply]
Without dangerously skip they are serializing as an escaped string as expected. You definitely want to skip escaping. 

This is all `vite-plugin-ssr` specific behavior and has nothing to do with the browser. I'm unclear why dangerously skip is doing HTML validation maybe it's something they are doing downstream? For instance here is it working in SolidStart: https://hackernews.ryansolid.workers.dev/

In any case it isn't hard for me to make that comment not use the shorthand.
